app/models/step_by_step_page.rb

```diff
@@ -158,6 +158,10 @@ def discard_notes
     internal_change_notes.where(edition_number: nil).delete_all
   end
 
+  def can_be_claimed_for_2i?
+    status.submitted_for_2i?
```

Comment:
I'd avoid creating a method just for this, and call `step_by_step_page.status.submitted_for_2i?` directly.

Reply:
Yes, I've already changed that. It looks like this as I had to move all the other chaining conditions (which required the `current_user` out of this).